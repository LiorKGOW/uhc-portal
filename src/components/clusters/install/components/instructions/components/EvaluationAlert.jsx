import React from 'react';
import {
  Alert,
} from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import links from '../../../../../../common/installLinks';

function EvaluationAlert() {
  const title = 'New clusters are automatically registered with a 60-day evaluation subscription.';
  const description = (
    <>
      Evaluation subscriptions do not include support from Red Hat.
      For non-evaluation use, you should attach a subscription that includes support.
      {' '}
      <a href={links.SUBSCRIPTION_EVAL_INFORMATION} target="_blank" rel="noreferrer noopener">
        Learn more
        {' '}
        <ExternalLinkAltIcon color="#0066cc" size="sm" />
      </a>
      .
    </>
  );

  return (
    <Alert variant="warning" isInline title={title}>
      {description}
    </Alert>
  );
}

export default EvaluationAlert;
