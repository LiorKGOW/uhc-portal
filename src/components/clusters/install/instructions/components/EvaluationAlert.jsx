import React from 'react';
import { Alert } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import links from '../../../../../common/installLinks.mjs';

function EvaluationAlert() {
  const title = 'New clusters are automatically registered with a 60-day evaluation subscription.';
  const description = (
    <>
      Evaluation subscriptions do not include support from Red Hat. You can edit your subscription
      settings after the cluster is created to receive support.{' '}
      <a href={links.SUBSCRIPTION_EVAL_INFORMATION} target="_blank" rel="noreferrer noopener">
        Learn more <ExternalLinkAltIcon color="#0066cc" size="sm" />
      </a>
      .
    </>
  );

  return (
    <Alert variant="warning" isInline title={title}>
      {description}
    </Alert>
  );
}

export default EvaluationAlert;
