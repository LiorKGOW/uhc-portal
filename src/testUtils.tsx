import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
