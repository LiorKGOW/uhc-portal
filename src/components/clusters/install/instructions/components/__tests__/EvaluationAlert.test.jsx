import React from 'react';

import { checkAccessibility, render } from '~/testUtils';

import EvaluationAlert from '../EvaluationAlert';

describe('<EvaluationAler />', () => {
  it('is accessible', async () => {
    const { container } = render(<EvaluationAlert />);

    await checkAccessibility(container);
  });
});
