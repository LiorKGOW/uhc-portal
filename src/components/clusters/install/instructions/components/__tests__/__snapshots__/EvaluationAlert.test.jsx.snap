// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluationAlert renders correctly 1`] = `
<Alert
  isInline={true}
  title="New clusters are automatically registered with a 60-day evaluation subscription."
  variant="warning"
>
  Evaluation subscriptions do not include support from Red Hat. You can edit your subscription settings after the cluster is created to receive support.
   
  <a
    href="https://access.redhat.com/articles/4389911"
    rel="noreferrer noopener"
    target="_blank"
  >
    Learn more 
    <ExternalLinkAltIcon
      color="#0066cc"
      noVerticalAlign={false}
      size="sm"
    />
  </a>
  .
</Alert>
`;
